src/components/AstronautsPage/reducers/astronautsReducer.ts

```diff
@@ -0,0 +1,39 @@
+import { AnyAction } from '@reduxjs/toolkit';
+import { AstronautsActions } from '../actions/actionTypes';
+
+const InitialState = {
+  astronauts: [],
+  isLoading: false,
+  error: null,
+};
+
+export const astronautsReducer = (
+  state = InitialState,
+  { type, payload }: AnyAction,
+) => {
+  switch (type) {
+    case AstronautsActions.FETCH_ASTRONAUTS_REQUEST:
+      return {
+        ...state,
+        astronauts: [],
+        isLoading: true,
+        error: null,
+      };
+    case AstronautsActions.FETCH_ASTRONAUTS_SUCCESS:
+      return {
+        ...state,
+        astronauts: payload,
+        isLoading: false,
+        error: null,
+      };
+    case AstronautsActions.FETCH_ASTRONAUTS_ERROR:
+      return {
+        ...state,
+        astronauts: [],
+        isLoading: false,
+        error: payload.error,
```

Comment:
you don't repeat fields that you have in the initial state

Reply:
clear